web/web.go

```diff
@@ -0,0 +1,752 @@
+	case "mysql":
+		dataSourceName = fmt.Sprintf("%s:%s@tcp(%s:%s)/%s", config.User, config.Password, config.Host, config.Port, config.Database)
+	default:
+		http.Error(w, fmt.Sprintf("Driver : '%s' is not supported", config.Driver), http.StatusBadRequest)
+		return
+	}
+	sourceDB, err := sql.Open(config.Driver, dataSourceName)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("SQL connection error : %v", err), http.StatusInternalServerError)
+		return
+	}
+	// Open doesn't open a connection. Validate DSN data:
+	err = sourceDB.Ping()
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Connection Error: %v. Check Configuration again.", err), http.StatusInternalServerError)
+		return
+	}
+	app.sourceDB = sourceDB
+	app.dbName = config.Database
+	app.driver = config.Driver
+	w.WriteHeader(http.StatusOK)
+}
+
+func convertSchemaSQL(w http.ResponseWriter, r *http.Request) {
+	if app.sourceDB == nil || app.dbName == "" || app.driver == "" {
+		http.Error(w, fmt.Sprintf("Database is not configured or Database connection is lost. Please set configuration and connect to database."), http.StatusNotFound)
+		return
+	}
+	conv := internal.MakeConv()
+	var err error
+	switch app.driver {
+	case "mysql":
+		err = mysql.ProcessInfoSchema(conv, app.sourceDB, app.dbName)
+	case "postgres":
+		err = postgres.ProcessInfoSchema(conv, app.sourceDB)
+	default:
+		http.Error(w, fmt.Sprintf("Driver : '%s' is not supported", app.driver), http.StatusBadRequest)
+		return
+	}
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Schema Conversion Error : %v", err), http.StatusNotFound)
+		return
+	}
+	app.conv = conv
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(conv)
+}
+
+func convertSchemaDump(w http.ResponseWriter, r *http.Request) {
+	reqBody, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Body Read Error : %v", err), http.StatusInternalServerError)
+		return
+	}
+	var dc DumpConfig
+	err = json.Unmarshal(reqBody, &dc)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Request Body parse error : %v", err), http.StatusBadRequest)
+		return
+	}
+	f, err := os.Open(dc.FilePath)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to open the test data file: %v", err), http.StatusNotFound)
+		return
+	}
+	conv, err := conversion.SchemaConv(dc.Driver, &conversion.IOStreams{In: f, Out: os.Stdout}, 0)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Schema Conversion Error : %v", err), http.StatusNotFound)
+		return
+	}
+	app.conv = conv
+	app.driver = dc.Driver
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(conv)
+}
+
+func getDDL(w http.ResponseWriter, r *http.Request) {
+	c := ddl.Config{Comments: true, ProtectIds: false}
+	var tables []string
+	for t := range app.conv.SpSchema {
+		tables = append(tables, t)
+	}
+	sort.Strings(tables)
+	ddl := make(map[string]string)
+	for _, t := range tables {
+		ddl[t] = app.conv.SpSchema[t].PrintCreateTable(c)
+	}
```

Comment:
Calling app.conv.GetDDL() will print the whole ddl including foreign key statements, which are not handled in frontend currently, so we are skipping those statements for now, i'll add a TODO comment for that.

Reply:
I think this will be resolved as part of pr101: once that is submitted, we should be able to call directly into code in convert.go.

Reply:
One challenge here is that frontend requires a map to show these tables in dropdown, which has table name as title and when you expand it shows CREATE TABLE statement. but  app.conv.GetDDL() will return ddl in list. thats why we need to create this as a separate function instead.

Reply:
We discussed this a bit internally and decided that it might be fair to leave it as it is since this is only used to populate the "Show DDL" tab in UI. (Download ddl json on the other hand correctly calls the app.conv.GetDDL()). 

We should though expand the TODO a bit to mention these caveats, and should mention missing support for foreign keys (in addition to secondary indexes).

Reply:
How about a comment along the lines of:

// getDDL returns the Spanner DDL for each table in alphabetical order.
// Unlike internal/convert.go's GetDDL, it does not print tables in a way that
// respects the parent/child ordering of interleaved tables. This means that
// getDDL cannot be used to build DDL to send to Spanner.

Reply:
added comment as mentioned